broadcastclient/broadcastclient_test.go

```diff
@@ -52,32 +63,52 @@ func TestReceiveMessages(t *testing.T) {
 }
 
 type dummyTransactionStreamer struct {
+	t               *testing.T
 	messageReceiver chan broadcaster.BroadcastFeedMessage
+	chainId         uint64
+	sequencerAddr   *common.Address
 }
 
-func NewDummyTransactionStreamer() *dummyTransactionStreamer {
+func NewDummyTransactionStreamer(t *testing.T, chainId uint64, sequencerAddr *common.Address) *dummyTransactionStreamer {
 	return &dummyTransactionStreamer{
 		messageReceiver: make(chan broadcaster.BroadcastFeedMessage),
+		chainId:         chainId,
+		sequencerAddr:   sequencerAddr,
 	}
 }
 
-func (ts *dummyTransactionStreamer) AddBroadcastMessages(pos arbutil.MessageIndex, messages []arbstate.MessageWithMetadata) error {
-	for i, message := range messages {
-		ts.messageReceiver <- broadcaster.BroadcastFeedMessage{
-			SequenceNumber: pos + arbutil.MessageIndex(i),
-			Message:        message,
+func (ts *dummyTransactionStreamer) AddBroadcastMessages(ctx context.Context, feedMessages []*broadcaster.BroadcastFeedMessage) error {
+	for _, feedMessage := range feedMessages {
+		if !ts.IsValidSignature(ctx, feedMessage) {
+			ts.t.Log("feedMessage invalid signature")
 		}
+		ts.messageReceiver <- *feedMessage
 	}
 	return nil
 }
 
+func (ts *dummyTransactionStreamer) IsValidSignature(_ context.Context, message *broadcaster.BroadcastFeedMessage) bool {
```

Comment:
Do we need to duplicate all of this from `TransactionStreamer`? Looks like this test doesn't do anything with signing so we could probably skip it

Reply:
removed as part of refactor